app/logic/events.py

```diff
@@ -118,11 +110,19 @@ def getStudentLedProgram(term):
 
 def getTrainingProgram(term):
 
-    trainingEvents = (Event.select(Event, Program.id.alias("program_id"))
-                           .join(ProgramEvent)
-                           .join(Program)
+    allTrainingsEvent = ((ProgramEvent.select(ProgramEvent.event, fn.COUNT(1).alias('num_programs'))
+                                        .join(Event)
+                                        .group_by(ProgramEvent.event)
+                                        .order_by(fn.COUNT(1).desc()))
+                                        .where(Event.term == term).get())
```

Comment:
I'd put a comment above this query describing how it works, since some students won't be familiar with the aggregate functions